src/functions.rs

```diff
@@ -2827,23 +2895,99 @@ fn validate_argument<'a>(
             if arg_typeinfo.is_child_or_actual_type(target_argument.param_type, types) {
                 Ok(TypeInstance::new(&arg, arg_typeinfo, file, context))
             } else {
-                Err(ErrorItem::make_compile_or_internal_error(
-                    &format!(
-                        "Expected type inheriting {}",
-                        target_argument.param_type.name
-                    ),
+                Err(validate_argument_error_handler(
+                    arg,
+                    arg_typeinfo,
+                    target_argument,
+                    types,
                     file,
-                    arg.get_range(),
-                    &format!(
-                        "This type should inherit {}",
-                        target_argument.param_type.name
-                    ),
+                    func_info,
                 ))
             }
         }
     }
 }
 
+// Helper function for handling the error case for validate_argument.
+fn validate_argument_error_handler(
+    arg: ArgForValidation,
+    arg_typeinfo: &TypeInfo,
+    target_argument: &FunctionArgument,
+    types: &TypeMap,
+    file: Option<&SimpleFile<String, String>>,
+    func_info: Option<&FunctionInfo>,
+) -> ErrorItem {
+    // If there is no function call or the call is not an asociated one fall through and return
```

Comment:
I don't think "no function call" accurately describes the situation.  And it's mildly confusing, since why are we validating no function.

The situation is that built-ins don't have FunctionInfos, so if func_info is None, that means we're validating a built-in, which can't be an associated call.

Please reword the comment to clarify that None = builtin, which implies not an associated_call.